remixer_storage/src/main/java/com/google/android/libraries/remixer/storage/json/ValueConverter.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.libraries.remixer.storage.json;
+
+import com.google.android.libraries.remixer.RemixerItem;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import java.util.ArrayList;
+
+/**
+ * Helper object that abstracts the type-dependent parts of parsing a Variable from Json and
+ * converting it from a java object into its JSON representation.
+ * @param <T> the data type stored in the variable.
+ */
+abstract class ValueConverter<T> {
+
+  /**
+   * The data type this converter is used for.
+   */
+  private final String dataType;
+
+  ValueConverter(String dataType) {
+    this.dataType = dataType;
+  }
+
+  /**
+   * Returns an object of type T that holds the value in the current Json Element.
+   */
+  abstract T parseValue(JsonElement element);
+
+  /**
+   * Returns a JsonElement that represents the value passed in.
+   */
+  abstract JsonElement valueToJson(T value);
+
+  /**
+   * Instantiates an appropriate {@link StoredVariable} for the data type.
+   */
+  abstract StoredVariable<T> createStoredVariable();
+
+  /**
+   * Creates a StoredVariable that represents the data in {@code item} if {@code item} is of this
+   * type. Returns null otherwise.
+   */
+  abstract StoredVariable<T> fromRemixerItem(RemixerItem item);
+
+  /**
+   * Deserializes a JsonElement that contains a StoredVariable.
+   */
+  StoredVariable<T> deserialize(JsonElement json) {
+    StoredVariable<T> result = createStoredVariable();
```

Comment:
Done. This was refactored from before the deserialization had the <T> param, so it just stayed there accidentally, thanks